ui/library/src/utils/ScrollDirectionDetector.ts

```diff
@@ -52,6 +75,23 @@ export default class ScrollDetector {
     if (this._lastScrollDirection !== currScrollDirection) {
       this._lastScrollDirection = currScrollDirection;
       this._setScrollDirection(currScrollDirection);
+
+      this._lastScrollTopBeforeDirectionChange = currScrollTop;
+      this._isScrollThresholdReachedInDirection = null;
+      if (this._setScrollThresholdReachedInDirection !== undefined && !this._isAtTop) {
```

Comment:
@ericmarsh995 technically u can just do `if (this._setScrollThresholdReachedInDirection && !this._isAtTop)`